Warning when compiling diff.c

### Steps to reproduce

Start make, CONF_OPT_FEAT = --with-features=normal

### Expected behaviour

No warnings from the compiler

### Version of Vim

8.2.3941

### Environment

Debian, gcc 11.2.0

### Logs and stack traces

```shell
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -O2 -fno-strength-reduce -Wall -Wno-deprecated-declarations -D_REENTRANT -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1        -o objects/diff.o diff.c
diff.c: In function ‘diff_try_update’:
diff.c:3284:26: warning: ‘hunk’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 3284 |         hunk->count_orig = oldcount;
      |         ~~~~~~~~~~~~~~~~~^~~~~~~~~~
diff.c:1639:18: note: ‘hunk’ was declared here
 1639 |     diffhunk_T  *hunk;
      |                  ^~~~
```

--- a/src/nvim/diff.c
+++ b/src/nvim/diff.c
@@ -1536,7 +1536,7 @@ static void diff_read(int idx_orig, int idx_new, diffio_T *dio)
   long off;
   int i;
   int notset = true;  // block "*dp" not set yet
-  diffhunk_T *hunk;
+  diffhunk_T *hunk = NULL;  // init to avoid gcc warning
   enum {
     DIFF_ED,
     DIFF_UNIFIED,

